src/unix.rs

```diff
@@ -67,9 +40,23 @@ mod tests {
     }
 
     #[test]
-    fn invalid_locale_doesnt_crash() {
-        std::env::set_var("LANG", "invalid");
-        std::env::set_var("LC_ALL", "invalid-again");
+    fn env_priority() {
+        env::remove_var(LANG);
```

Comment:
The current discussion around manipulating the environment seems to lean towards this being non-ideal. In some other code I've seen, tests like this were changed to use an "inner" method of `get()` that took a `HashMap<String, String>` so that it could be mocked as necessary without possibly racing with other tests down the line.

Do you think something like that could work here too?